tornado/iostream.py

```diff
@@ -1006,12 +1002,6 @@ def _consume(self, loc: int) -> bytes:
         ).tobytes()
         self._read_buffer_pos += loc
         self._read_buffer_size -= loc
-        # Amortized O(1) shrink
-        # (this heuristic is implemented natively in Python 3.4+
```

Comment:
What this means (IIRC) is that `del self._read_buffer[:self._read_buffer_pos]` is amortized O(1) in Python 3.4+ (but is less efficient in older versions). So what we need to do here is not delete the code completely but just make it unconditional. 

And then once we are always resetting `_read_buffer_pos` to zero, maybe we can remove that attribute completely.

Reply:
Thanks for your explanation. Advice applied.